### PR TITLE
[parallel] feat: add enable_forward_prefetch for fsdp2

### PR DESCRIPTION
**Pull request by JorgenWan**
### What does this PR do?

> Concise overview of the change. Reference related issues/PRs.

### Checklist Before Starting

- Search for relative PRs/issues and link here: ...
- PR title follows `[{modules}] {type}: {description}` format (enforced by [check_pr_title.yml](.github/workflows/check_pr_title.yml))
  - **Allowed modules:** `agent`, `ci`, `ckpt`, `config`, `data`, `dist`, `docker`, `docs`, `logging`, `misc`, `model`, `omni`, `optim`, `ops`, `parallel`, `perf`, `release`, `task`, `trainer`
  - **Allowed types:** `feat`, `fix`, `refactor`, `chore`, `test`
  - Breaking changes: prepend `[BREAKING]` — e.g. `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> Validation results (training curves, eval metrics) for changes not covered by CI.

### API and Usage Example

> Show API changes and usage examples if applicable.

### Design & Code Changes

> High-level design description and specific change list.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

### CODE REVIEW

File: veomni/distributed/torch_parallelize.py [hunk 1]
```diff
@@ -532,7 +532,9 @@ def parallelize_model_fsdp2(
     fully_shard(model, **fsdp_kwargs)
 
     # configure manual prefetching when needed
-    need_manual_prefetch = parallel_state.any_extra_parallel_enabled or mp_ignored_classes is not None
+    need_manual_prefetch = (
+        parallel_state.any_extra_parallel_enabled or mp_ignored_classes is not None
+    ) and kwargs.pop("enable_forward_prefetch", True)
```
[the remark this answers is not in the document]
Comment (JorgenWan) [reply veomni/distributed/torch_parallelize.py:537]: kwargs uses enable_forward_prefetch as fsdp1

**Review by Coach257 — APPROVED** [review]
